### PR TITLE
Add uWSGI input plugin

### LINKED ISSUE #920: Feature request: uWSGI input plugin.
uWSGI can be configured to [expose stats](http://uwsgi-docs.readthedocs.org/en/latest/StatsServer.html).

### PR DESCRIPTION
**Pull request by OrangeTux**
Fixes #920

### CODE REVIEW

File: plugins/inputs/uwsgi/uwsgi.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+func (u *Uwsgi) Gather(acc telegraf.Accumulator) error {
+	for _, s := range u.Servers {
+		n, err := url.Parse(s)
+		if err != nil {
+			return fmt.Errorf("Could not parse uWSGI Stats Server url '%s': %s", s, err)
+		}
+
+		u.gatherServer(acc, n)
```
Comment (sparrc) [inline plugins/inputs/uwsgi/uwsgi.go]: you shouldn't ignore the error here, you should accumulate any errors from servers that fail and then return that.

File: plugins/inputs/uwsgi/uwsgi_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
+package uwsgi_test
```
Comment (sparrc) [inline plugins/inputs/uwsgi/uwsgi_test.go]: please also write a test checking that errors are returned from Gather properly. You can just use invalid addresses for http, tcp, unix to test that.

File: plugins/inputs/uwsgi/uwsgi.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	if err != nil {
+		return fmt.Errorf("Could not connect to uWSGI Stats Server '%s': %s", url.String(), err)
+	}
+	defer r.Close()
+
+	var s StatsServer
+	s.Url = url.String()
+
+	dec := json.NewDecoder(r)
+	dec.Decode(&s)
```
Comment (sparrc) [inline plugins/inputs/uwsgi/uwsgi.go]: don't ignore errors here either

File: plugins/inputs/uwsgi/uwsgi_test.go (hunk 2 above)
Comment (sparrc) [inline plugins/inputs/uwsgi/uwsgi_test.go]: also write a test with an http returning invalid JSON, to make sure that an error is returned there as well

**Comment by sparrc** [comment]
Thanks @OrangeTux, overall it looks good but just some small changes, mostly around error handling and testing

File: plugins/inputs/uwsgi/uwsgi.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+	Servers []string `toml:"server"`
+}
+
+func (u *Uwsgi) Description() string {
+	return "Read uWSGI metrics."
+}
+
+func (u *Uwsgi) SampleConfig() string {
+	return `
+    ## List with urls of uWSGI Stats servers. Url must match pattern:
```
Comment (sparrc) [inline plugins/inputs/uwsgi/uwsgi.go]: these should be indented 2 spaces

**Comment by sparrc** [comment]
@OrangeTux are you able to make the requested changes?

**Comment by sparrc** [comment]
closing for lack of updates.

@OrangeTux Please keep in mind that if you submit PRs you are going to need to follow-up with revisions in a reasonable amount of time.

**Comment by blaggacao** [comment]
@OrangeTux are you still working on this? Some people I think out there are starving for uwsgi support... :smile:

**Pull request closed without merging**